rpcserver.go

```diff
@@ -473,6 +477,50 @@ func (r *rpcServer) listBalancesByFamilyKey(ctx context.Context,
 	return resp, nil
 }
 
+// ListUtxos lists the UTXOs managed by the target daemon, and the assets they
+// hold.
+func (r *rpcServer) ListUtxos(ctx context.Context,
+	_ *tarorpc.ListUtxosRequest) (*tarorpc.ListUtxosResponse, error) {
+
+	rpcAssets, err := r.fetchRpcAssets(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	managedUtxos, err := r.cfg.AssetStore.FetchManagedUTXOs(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	utxos := make(map[string]*tarorpc.ManagedUtxo)
+	for _, u := range managedUtxos {
+		utxos[u.OutPoint.String()] = &tarorpc.ManagedUtxo{
+			OutPoint:    u.OutPoint.String(),
+			AmtSat:      int64(u.OutputValue),
+			InternalKey: u.InternalKey.PubKey.SerializeCompressed(),
+			TaroRoot:    u.TaroRoot,
+		}
+	}
+
+	// Populate the assets managed by each UTXO.
+	for _, a := range rpcAssets {
+		op := a.ChainAnchor.AnchorOutpoint
+		utxo, ok := utxos[op]
+		if !ok {
+			return nil, fmt.Errorf("unable to find utxo %s for "+
+				"asset_id=%s", op,
+				string(a.AssetGenesis.AssetId))
```

Comment:
I think this should be `asset_id=%x` and no conversion to string. Otherwise it will print the raw bytes as ASCII characters.

Reply:
Done: https://github.com/lightninglabs/taro/pull/185/commits/d72ceae415c9706dbd102a87d4d0bfc26594965f